File: models/assetmanagement/AssetMaintenance.js
import mongoose from "mongoose";

const AssetMaintenanceSchema = new mongoose.Schema(
{
    hospital_id: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branch_id: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    asset_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "AssetMaster"
    },

    maintenance_type: {
        type: String,
        enum: ["preventive","repair","breakdown"]
    },

    technician_name: String,

    vendor_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "AssetVendor"
    },

    maintenance_date: Date,

    cost: Number,

    next_maintenance_date: Date,

    description: String

},
{ timestamps: true }
);

export default mongoose.model("AssetMaintenance", AssetMaintenanceSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/AssetMaintenance.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "asset_maintenances")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AssetMaintenance extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "asset_id")
    private AssetMaster asset;

    @Enumerated(EnumType.STRING)
    @Column(name = "maintenance_type")
    private MaintenanceType maintenanceType;

    @Column(name = "technician_name")
    private String technicianName;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vendor_id")
    private AssetVendor vendor;

    @Column(name = "maintenance_date")
    private LocalDateTime maintenanceDate;

    private Double cost;

    @Column(name = "next_maintenance_date")
    private LocalDateTime nextMaintenanceDate;

    @Enumerated(EnumType.STRING)
    private Status status = Status.scheduled;

    private String description;

    public enum MaintenanceType {
        preventive, repair, breakdown
    }

    public enum Status {
        scheduled, in_progress, completed
    }
}
